src/pages/selfhosted/identity-providers.mdx

```diff
@@ -960,7 +960,157 @@ NETBIRD_AUTH_DEVICE_AUTH_SCOPE="openid email"
+</p>
+- Take note of `Client ID` and `Client Secret` and click `OK`
+<p>
+    <img src="/docs-static/img/integrations/identity-providers/self-hosted/google-oauth-client-created.png" alt="high-level-dia" class="imagewrapper"/>
+</p>
+
+### Step 3: Create service account
+- Navigate to [API Credentials](https://console.cloud.google.com/apis/credentials) page
+- Click `CREATE CREDENTIALS` at the top and select `Service account`
+- Fill in the form with the following values and click `CREATE`
+    - Service account name: `netbird`
+    - Service account ID: `netbird`
+- Take note of service account email address, we will use it later
+- Click `DONE`
+<p>
+    <img src="/docs-static/img/integrations/identity-providers/self-hosted/google-service-account-create.png" alt="high-level-dia" class="imagewrapper"/>
+</p>
+
+### Step 4: Grant user and schema management admin role to service account
+- Navigate to [Admin Console](https://admin.google.com/ac/home) page
+- Select `Account` on the left menu and then click `Admin Roles`
+- Click `Create new role`
+- Fill in the form with the following values and click `CREATE`
+    - name: `User and Schema Management Admin`
+    - description: `User and Schema Management Admin`
+- Click `CONTINUE`
+<p>
+    <img src="/docs-static/img/integrations/identity-providers/self-hosted/google-new-role-info.png" alt="high-level-dia" class="imagewrapper"/>
+</p>
+
+- Scroll down to `Admin API privileges` and add the following provileges
+    - Users: `Create`, `Read` and  `Update Custom Attributes`
+    - Schema Management: `Schema Read`
+- Click `CONTINUE`
+<p>
+    <img src="/docs-static/img/integrations/identity-providers/self-hosted/google-privileges-review.png" alt="high-level-dia" class="imagewrapper"/>
+</p>
+- Verify preview of assigned Admin API privileges to ensure that everything is properly configured, and then click `CREATE ROLE`
+- Click `Assign service accounts`, add service account email address and then click `ADD`
+<p>
+    <img src="/docs-static/img/integrations/identity-providers/self-hosted/google-assign-role.png" alt="high-level-dia" class="imagewrapper"/>
+</p>
+- Click `ASSIGN ROLE` to assign service account to `User and Schema Management Admin` role
+<p>
+    <img src="/docs-static/img/integrations/identity-providers/self-hosted/google-service-account-privileges.png" alt="high-level-dia" class="imagewrapper"/>
+</p>
+- Open downloaded json file and take note of `client_id` will be used later as `Service Account Client ID`
+
+### Step 5: Granting service account access to organization data
+- Navigate to [Admin Console](https://admin.google.com/ac/home) page
+- Select `Security` > `Access and data control` > `API controls` and then click `MANAGE DOMAIN WIDE DELEGATION`
+- Click `Add new`
+- Fill in the form with the following values
+    - Client ID: `<Service Account Client ID>`
```

Comment:
We can mention is step 3 also to write the service account ID down (we only mention email)